gpustack/worker/model_file_manager.py

```diff
@@ -95,10 +102,112 @@ async def _handle_deletion(self, model_file: ModelFile):
                 logger.info(
                     f"Cancelled download for deleted model: {model_file.readable_source}(id: {model_file.id})"
                 )
+                # Make sure to clean it up
+                if model_file.cleanup_on_delete:
+                    await self._delete_model_file(model_file)
 
-        if model_file.cleanup_on_delete:
+        elif not entry and model_file.cleanup_on_delete:
             await self._delete_model_file(model_file)
```

Comment:
How does it differ from previous implementation? Is there any other exceptions other than the CancelledError?